integration/container/stop_test.go

```diff
@@ -42,6 +43,47 @@ func TestStopContainerWithRestartPolicyAlways(t *testing.T) {
 	}
 }
 
+// TestStopContainerWithTimeout checks that ContainerStop with
+// a timeout works as documented, i.e. in case of negative timeout
+// waiting is not limited (issue #35311).
+func TestStopContainerWithTimeout(t *testing.T) {
+	t.Parallel()
```

Comment:
This makes for a flakey test whennused with the main integration daemon.

Reply:
Can you please elaborate why?

Reply:
Because it is running in the main daemon.
The test setup takes a snapshot of the current state and then at the end of the test cleans up anything that was not in the original state. But if there are multiple tests doing this cleanup they will conflict with each other, clean up stuff they shouldn't be, etc.